src/p2p/Lost.ts

```diff
@@ -1192,8 +1192,101 @@ async function isDownCheck(node) {
+      let hash = crypto.hash(obj)
+      let hashTrieReq: HashTrieReq = {
+        radixList: ['0'],
+      }
+      let closestNodes = stateManager.getClosestNodes(hash, 5, true)
+      let nearNode: P2P.NodeListTypes.Node
+      for (let closetNode of closestNodes) {
+        if (closetNode.id !== node.id) {
+          nearNode = closetNode
+          break
+        }
+      }
+      if (nearNode == null) {
+        throw new Error(`isDownCheck unable to get near node to check`)
+      }
+
+      //check radix response for near node
+      let nearNodeAsk
+      if (
+        this.stateManager.config.p2p.useBinarySerializedEndpoints &&
+        this.stateManager.config.p2p.getTrieHashesBinary
+      ) {
+        nearNodeAsk = Comms.askBinary<GetTrieHashesRequest, GetTrieHashesResponse>(
+          nearNode,
+          InternalRouteEnum.binary_get_trie_hashes,
+          hashTrieReq,
+          serializeGetTrieHashesReq,
+          deserializeGetTrieHashesResp,
+          {}
+        )
+      } else {
+        nearNodeAsk = this.p2p.ask(node, 'get_trie_hashes', hashTrieReq)
+      }
+      const nearNodeResponse = await nearNodeAsk
+
+      if (logFlags.verbose)
+        info(
+          `lost check result for nearNodeAsk near node:${node.id} cycle ${currentCycle} is ${utils.stringifyReduce(
+            nearNodeResponse
+          )}`
+        )
+
+      if (nearNodeResponse == null) {
+        /* prettier-ignore */ nestedCountersInstance.countEvent('p2p', 'isDownCheck-down-near-node-fail', 1)
+        return 'down'
+      }
+
+      let targetNodeAsk
+      if (
+        this.stateManager.config.p2p.useBinarySerializedEndpoints &&
+        this.stateManager.config.p2p.getTrieHashesBinary
+      ) {
+        targetNodeAsk = Comms.askBinary<GetTrieHashesRequest, GetTrieHashesResponse>(
+          nearNode,
+          InternalRouteEnum.binary_get_trie_hashes,
+          hashTrieReq,
+          serializeGetTrieHashesReq,
+          deserializeGetTrieHashesResp,
+          {}
+        )
+      } else {
+        targetNodeAsk = this.p2p.ask(node, 'get_trie_hashes', hashTrieReq)
+      }
+      const targetNodeResponse = await targetNodeAsk
+
+      if (logFlags.verbose)
+        info(
+          `lost check result for nearNodeAsk target node:${node.id} cycle ${currentCycle} is ${utils.stringifyReduce(
+            targetNodeResponse
+          )}`
+        )
+
+      if (targetNodeResponse == null) {
+        /* prettier-ignore */ nestedCountersInstance.countEvent('p2p', 'isDownCheck-down-near-node-fail', 1)
+        return 'down'
+      }
+
+      const nodeHashesMap = new Map<string, string>()
+      for (let nodeHash of nearNodeResponse.nodeHashes) {
+        nodeHashesMap.set(nodeHash.radix, nodeHash.hash)
+      }
+      //compare the hashes from the near node and the target node
+      for (let nodeHash of targetNodeResponse.nodeHashes) {
+        if (nodeHashesMap.get(nodeHash.radix) !== nodeHash.hash) {
```

Comment:
What if the target node is up and responsive but has out of sync data? In that case, we should not mark it  as `down` as the node may repair the radix in next cycle.